src/MagnumPlugins/BasisImporter/Test/BasisImporterTest.cpp

```diff
@@ -0,0 +1,246 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2019 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNETCION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <sstream>
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/TestSuite/Tester.h>
+#include <Corrade/Utility/ConfigurationGroup.h>
+#include <Corrade/Utility/DebugStl.h>
+#include <Corrade/Utility/Directory.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Trade/AbstractImporter.h>
+#include <Magnum/Trade/ImageData.h>
+
+#include "configure.h"
+
+namespace Magnum { namespace Trade { namespace Test { namespace {
+
+struct BasisImporterTest: TestSuite::Tester {
+    explicit BasisImporterTest();
+
+    void empty();
+    void invalid();
+    void unconfigured();
+
+    void openTwice();
+    void importTwice();
+
+    void rgb();
+    void rgba();
+
+    /* Explicitly forbid system-wide plugin dependencies */
+    PluginManager::Manager<AbstractImporter> _manager{"nonexistent"};
+};
+
+constexpr struct {
+    const char* suffix;
+    const char* expectedFormatStr;
+    const CompressedPixelFormat expectedFormat;
+    const CompressedPixelFormat expectedFormatAlpha;
+    const bool needsPow2;
```

Comment:
This again leads to `?:` in the test, allowing just two hardcoded variants. Would be better as 

```cpp
const char* file;
Vector2i expectedSize;
```

as it allows more parametrization in the future ;)